src/validation.cpp

```
@@ -3787,6 +3788,29 @@ static FILE* OpenDiskFile(const CDiskBlockPos &pos, const char *prefix, bool fRe
     fs::path path = GetBlockPosFilename(pos, prefix);
     fs::create_directories(path.parent_path());
     FILE* file = fsbridge::fopen(path, fReadOnly ? "rb": "rb+");
+    if (!file && has_master_datadir) {
+        // fall back to master source file; we copy it if !fReadOnly, otherwise
+        // we just open it as is (presuming it exists)
+        fs::path mpath = GetBlockPosFilename(pos, prefix, true);
+        FILE* mfile = fsbridge::fopen(mpath, "rb");
+        if (mfile) {
+            if (fReadOnly) {
+                // good as is
+                return mfile;
+            }
+            // need writing; copy to own datadir
+            file = fsbridge::fopen(path, "wb+");
+            uint8_t buffer[16384];
+            while (!feof(mfile)) {
+                size_t r = fread(buffer, 1, 16384, mfile);
+                if (r) fwrite(buffer, 1, r, file);
```

Should handle failure instead of ignoring? Could lead to corrupt blocks?